src/entities/Seed.ts

```diff
@@ -374,6 +374,12 @@ export class Seed {
     }
   }
 
+  async fundSeed(): Promise<TransactionReceipt> {
+    return this.transactionsService.send(
+      () => this.projectTokenContract.transfer(this.contract.address, this.seedAmountRequired)
```

Comment:
for all of these new methods, can you please be sure you update the state of the seed similar to what you see being done in the `buy` method?  Note: don't hydrate user info unless needed.

Reply:
okay, I'll do that.

Reply:
have done, but I have some doubts regarding how to update state (Have updated the code).
1. for `pause`, `unPause` and `close`, I should use `hydateClosedOrPaused` method provided in Seed entity, right?
2. for `retrieveProjectTokens` and `withdrawFundingTokens`, I should use `hydrateTokensState` method provided in Seed entity, right?
3. for functions related to whitelisting, should I update the state? sometime, admin might later add himself/herself to the whitelist, do we want to update the user info too?

Reply:
I think it's done. Can you check it please.